### ISSUE TITLE
Marker fails to identify incorrect link if both link text and link have an underscore

### ISSUE THREAD

**Issue by AlexNPavel** [crawford/marker#16]
Title describes the problem. 

Only link has underscore:

```
This is a [linkTextWithUnderscore][link_with_underscore].
```

Output: 

```
Found broken reference (linkTextWithUnderscore -> linkTextWithUnderscore) at ./test.md:1
```

Only link text has underscore:

```
This is a [link_text_with_underscore][linkWithUnderscore].
```

Output:

```
Found broken reference (linkWithUnderscore -> linkWithUnderscore) at ./test.md:1
```

Both link text and link have underscore:

```
This is a [link_text_with_underscore][link_with_underscore].
```

Command succeeds.

**Comment by crawford** [comment]
This is due to https://github.com/raphlinus/pulldown-cmark/issues/146. Your example ends up being parsed as follows:

```
Text("This is a ")
Text("[link")
Text("_text")
Text("_with")
Text("_underscore]")
Text("[link")
Text("_with")
Text("_underscore].")
```

Marker is looking for a text event with both `[` and `]` and doesn't know that it can concatenate adjacent text events.

**Comment by crawford** [comment]
Fixed in c9da2c6bd50c06b0901ef4ffd37523ac4397bd70.
